test/baselines/adapter_TS_React/admin/src/components/settings.tsx

```diff
@@ -1,124 +1,136 @@
 import * as React from "react";
-import { composeObject, entries } from "alcalzone-shared/objects";
-
-export type OnSettingsChangedCallback = (newSettings: Record<string, unknown>) => void;
+import { withStyles } from "@material-ui/core/styles";
+import { CreateCSSProperties } from "@material-ui/core/styles/withStyles";
+import TextField from "@material-ui/core/TextField";
+import Input from "@material-ui/core/Input";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+
+import I18n from "@iobroker/adapter-react/i18n";
+
+const styles = (): Record<string, CreateCSSProperties> => ({
+	input: {
+		marginTop: 0,
+		minWidth: 400,
+	},
+	button: {
+		marginRight: 20,
+	},
+	card: {
+		maxWidth: 345,
+		textAlign: "center",
+	},
+	media: {
+		height: 180,
+	},
+	column: {
+		display: "inline-block",
+		verticalAlign: "top",
+		marginRight: 20,
+	},
+	columnLogo: {
+		width: 350,
+		marginRight: 0,
+	},
+	columnSettings: {
+		width: "calc(100% - 370px)",
+	},
+	controlElement: {
+		//background: "#d2d2d2",
+		marginBottom: 5,
+	},
+});
 
 interface SettingsProps {
-	onChange: OnSettingsChangedCallback;
-	settings: Record<string, unknown>;
+	classes: Record<string, string>;
+	native: any;
+
+	onChange: (attr: string, value: any) => void;
 }
 
-// Define here which settings exist and how they are typed
 interface SettingsState {
-	// catch-all for new and temporary values
-	[key: string]: unknown;
-	// they should all be optional, since they might not be set yet
-	// e.g. password?: string;
+	// add your state properties here
+	dummy?: undefined;
```

Comment:
should this just be `Partial<ioBroker.AdapterConfig>`?

Reply:
No, the state of the `<Settings>` is actually never used (the way Bluefox did this), the initial value is seeded through `props.native` and then editor changes are passed to the owner through `onChange()`.

But: `native` should probably be typed. What do you think? We could even make `GenericApp` generic with the adapter config type as type argument. Like that it wouldn't be `native: any` everywhere.

Reply:
Then we need no type for the state and no assignment `this.state = {}`.
`props.native` should be typed I think. And I like typing the config in `GenericApp`